### PR TITLE
WIP: allow kubernetes Secrets to be injected into the worker's environment

### PR DESCRIPTION
**Pull request by mmiller1**
As stated, this should probably be considered a WIP, it's only been implemented for postgres, and ideally it would let you 1) mount multiple secrets into the environment, and 2) mount Secrets into the worker's filesystem. this currently provides the functionality I need to use this bot in a more production'y manor right now, but I may circle back to do more work on it if time allows, and if it's not the totally wrong direction for this kind of thing that folks have envisioned for the project.

### CODE REVIEW

File: data/bundle.go [hunk 1]
```diff
@@ -109,6 +109,7 @@ type BundleCommand struct {
 // BundleKubernetes represents the "bundles/kubernetes" subsection of the config doc
 type BundleKubernetes struct {
 	ServiceAccountName string `yaml:"serviceAccountName,omitempty" json:"serviceAccountName,omitempty"`
+	EnvSecret          string `yaml:"env_secret,omitempty" json:"env_secret,omitempty"`
```
Comment (clockworksoul) [inline data/bundle.go:112]: Can `env_secret` be camelCased?

**Comment by clockworksoul** [comment]
Thanks for this PR! Apologies for not looking at it sooner: the real-life job, holidays, and a little case of the COVID have slowed me down a bit.

I'm absolutely not opposed to merging this almost as-is (barring the trivial casing request). It may be uprooted by the dynamic configuration changes I'm working on right now, but then again it's perfectly possible that it could stick around! The future is unclear.

Also, I think you got some bonus functionality out of this as well: I don't think that the in-memory DAL implementation actually needs any additional logic. This should work as-is for that as well.

**Comment by clockworksoul** [comment]
Also, once this merges, I'd be happy to cut a new version into main for you.

File: data/bundle.go (hunk 1 above)
Comment (clockworksoul) [reply data/bundle.go:112]: Actually: scratch that. It looks like all of the existing code is inconsistent with with the JSON standard, so proceed and I'll fix the entire codebase another time. :) 

Carry on. Nothing to see here.

**Comment by mmiller1** [comment]
Thanks, I'd appreciate a merge and new revision if being cut, if trivial.

**Comment by clockworksoul** [comment]
Can do.

**Pull request merged**